dnsimple/domains_pushes.go

```diff
@@ -0,0 +1,117 @@
+package dnsimple
+
+import (
+	"fmt"
+)
+
+// DomainPush represents a domain push in DNSimple.
+type DomainPush struct {
+	ID         int    `json:"id,omitempty"`
+	DomainID   int    `json:"domain_id,omitempty"`
+	ContactID  int    `json:"contact_id,omitempty"`
+	AccountID  int    `json:"account_id,omitempty"`
+	CreatedAt  string `json:"created_at,omitempty"`
+	UpdatedAt  string `json:"updated_at,omitempty"`
+	AcceptedAt string `json:"accepted_at,omitempty"`
+}
+
+// DomainPushAttributes represent a domain push payload (see initiate).
+type DomainPushAttributes struct {
+	NewAccountEmail string `json:"new_account_email,omitempty"`
+	ContactID       string `json:"contact_id,omitempty"`
+}
+
+// DomainPushResponse represents a response from an API method that returns a DomainPush struct.
+type DomainPushResponse struct {
+	Response
+	Data *DomainPush `json:"data"`
+}
+
+// DomainPushesResponse represents a response from an API method that returns a collection of DomainPush struct.
+type DomainPushesResponse struct {
+	Response
+	Data []DomainPush `json:"data"`
+}
+
+func initiateDomainPushPath(accountID string, domain interface{}) string {
+	return fmt.Sprintf("%v/pushes", domainPath(accountID, domain))
+}
+
+func domainPushPath(accountID string, pushID int) string {
+	path := fmt.Sprintf("%v/pushes", accountID)
+
+	if pushID != 0 {
+		path += fmt.Sprintf("/%d", pushID)
+	}
+
+	return path
+}
+
+// InitiatePush initiate a new domain push.
+//
+// See https://developer.dnsimple.com/v2/domains/pushes/#initiate
+func (s *DomainsService) InitiatePush(accountID string, domain interface{}, pushAttributes DomainPushAttributes) (*DomainPushResponse, error) {
```

Comment:
Instead of using a `domain interface{}`, I'd declare a `domainID string` as we have been doing lately. The `interface{}` is one of the remnants of legacy code that we'll likely have to clean sooner rather than later.